### PR TITLE
Optimizing each newspaper based on RSS feed

### PR DESCRIPTION
**Pull request by achaccha**
I branched parse logics for each newspapers because of RSS tag name difference between newspapers.
Please take a look and leave some comments for me.

### CODE REVIEW

**Review by royeom — APPROVED** [review]
How can i cancel approve ..? 

This is not approving , just mistake.

**Review by harrydrippin — COMMENTED** [review]
You need to change **all code** about `def extract()`. It seems you modified function signature, from 2 positional arguments to 3. So you also need to change `GithubCrawler`, not only `RSSCrawler`.

File: crawler/__init__.py
Comment (ethkim) [inline crawler/__init__.py]: Line break here to fit within 80 chars. (I think github's style guide is 80 for Python?)

File: crawler/__init__.py
Comment (ethkim) [inline crawler/__init__.py]: Same.

File: crawler/config.py
Comment (ethkim) [inline crawler/config.py]: Lifehacker  is one word (no space).

File: crawler/config.py
Comment (ethkim) [inline crawler/config.py]: Careful how? Please state exactly what needs to be done, and add a TODO.

e.g.

TODO(ethkim): Fix the XML feed consumption to prevent breaking from XXXX

File: crawler/config.py
Comment (ethkim) [inline crawler/config.py]: Where do we consume this mapping? We seem to have a lot of string-to-string mapping, where content provider names like "The Verge" is used in multiple structures (item_name, item_publish)...

File: crawler/rss.py
Comment (ethkim) [inline crawler/rss.py]: rename "temp" to something more meaningful. Each variable, method, etc need to be self-explanatory, and shouldn't require reading the code to understand what they do. 

Is each "temp" dictionary an article? Then let's call it article.

File: crawler/rss.py
Comment (ethkim) [inline crawler/rss.py]: We can merge this condition with the condition on line 33:

if item.name == None or item.name == RSSConfig.item_publish[code]:

File: crawler/rss.py
Comment (ethkim) [inline crawler/rss.py]: What is in item.name and item.string? I don't understand the comment on line 40.

File: test.py
Comment (ethkim) [inline test.py]: Remove these lines instead of just commenting out.

File: crawler/__init__.py
Comment (ethkim) [inline crawler/__init__.py]: Rename ret to something more meaningful.

**Review by ethkim — COMMENTED** [review]
Thanks! PTAL (Please take another look)

File: crawler/config.py
Comment (harrydrippin) [reply crawler/config.py]: ~~That issue was covered by commit "Optimize each newspaper". We'll delete that line.~~
We checked twice, and we find that issue was not covered. We'll modify the comment to TODO.

**Comment by harrydrippin** [comment]
@ethkim We added ~~2~~ 3 commits to fix some requested tasks. Please take a look.

File: crawler/config.py
Comment (ethkim) [inline crawler/config.py:14]: Line break after 80 chars. 

And TODO(@xxxx) should contain the person who added the TODO (and who will fix it later), not me. :)

File: crawler/config.py
Comment (ethkim) [inline crawler/config.py:18]: Are these items the name of sections for each content provider? Then we could say in the comment:

"Each item in the following list contains the name of sections (for each content provider), 
for which we extract content."

File: crawler/rss.py
Comment (ethkim) [inline crawler/rss.py:23]: extract -> remove.

File: crawler/rss.py
Comment (ethkim) [inline crawler/rss.py:46]: line break after 80 chars.

File: test.py
Comment (ethkim) [inline test.py:30]: ? Remove.

File: test.py
Comment (ethkim) [inline test.py:6]: This is ok for now, but: if this is meant to be a unit test, there should be unit testing framework. Please add a TODO to add that framework, and use mock data to test the crawl method rather than really running the Soup crawler. We should be testing extract() method, by passing in mock BeautifulSoup object.

**Review by ethkim — APPROVED** [review]
Thanks!! 

Approving this PR just to unblock you guys from merging, but please fix the comments before merging. :)

File: crawler/config.py
Comment (achaccha) [reply crawler/config.py:14]: I'll fix it.

File: crawler/config.py
Comment (achaccha) [reply crawler/config.py:18]: Ok! I'll fix it. Thank you for advice.

File: crawler/rss.py
Comment (achaccha) [reply crawler/rss.py:46]: I'll refactor it on next PR

File: test.py
Comment (achaccha) [reply test.py:6]: Now `unittest` module and some test codes are available on #2.

File: test.py
Comment (achaccha) [reply test.py:30]: OK. I'll remove `print`.

**Pull request merged**
